examples/cfd/aero_graph_net/inference.py

```diff
@@ -0,0 +1,181 @@
+        logger.info(f"Using {self.device} device")
+
+        # instantiate dataset
+        logger.info("Loading the test dataset...")
+        self.dataset = instantiate(cfg.data.test)
+        logger.info(f"Using {len(self.dataset)} test samples.")
+
+        # instantiate dataloader
+        logger.info("Creating the dataloader...")
+        self.dataloader = GraphDataLoader(
+            self.dataset,
+            **cfg.test.dataloader,
+        )
+
+        # instantiate the model
+        logger.info("Creating the model...")
+        self.model = instantiate(cfg.model).to(self.device)
+
+        # enable train mode
+        self.model.eval()
+
+        # load checkpoint
+        load_checkpoint(
+            to_absolute_path(cfg.resume_dir),
+            models=self.model,
+            device=self.device,
+        )
+
+        # instantiate losses.
+        logger.info("Creating the losses...")
+        self.loss = instantiate(cfg.loss)
+
+    @torch.no_grad()
+    def predict(self, save_results=False):
+        """
+        Run the prediction process.
+
+        Parameters:
+        -----------
+        save_results: bool
+            Whether to save the results in form of a .vtp file, by default False
+
+        Returns:
+        --------
+        None
+        """
+
+        for batch in self.dataloader:
+            batch = batch_as_dict(batch, self.device)
+            graph = batch["graph"]
+            # normals = normals.to(self.device, torch.float32).squeeze()
+            # areas = areas.to(self.device, torch.float32).squeeze()
+            # coeff = coeff.to(self.device, torch.float32).squeeze()
+            # sid = sid.item()
+            # self.logger.info(f"Processing sample ID {sid}")
+            pred = self.model(graph.ndata["x"], graph.edata["x"], graph)
+            gt = graph.ndata["y"]
+            pred, gt = self.dataset.denormalize(pred, gt, pred.device)
+
+            num_out_c = gt.shape[1]
+            if num_out_c in [1, 4]:
+                graph.ndata["p_pred"] = pred[:, 0]
+                graph.ndata["p"] = gt[:, 0]
+            if num_out_c in [3, 4]:
+                graph.ndata["wallShearStress_pred"] = pred[:, num_out_c - 3 :]
+                graph.ndata["wallShearStress"] = gt[:, num_out_c - 3 :]
+
+            error = self.loss.graph(pred, gt)
+            logger.info(f"Error (%): {error * 100:.4f}")
+
+            if save_results:
+                # Convert DGL graph to PyVista graph and save it
+                pv_graph = dgl_to_pyvista(graph.cpu())
+                assert (
+                    len(batch["name"]) == 1
+                ), "Only batch size 1 is currently supported."
+                pv_graph.save(self.output_dir / (batch["name"][0] + ".vtp"))
+
+
+@hydra.main(version_base="1.3", config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+    # initialize distributed manager
+    DistributedManager.initialize()
```

Comment:
After this: https://github.com/NVIDIA/modulus/pull/549, you can most likely remove...

Reply:
`initialize()` is still needed (dist manager is also used in logger init), but I updated the code to use the device from the manager on line 88.

Reply:
I see, and this is the logger you have defined in this PR correct?

Reply:
Yep - it's defined [here](https://github.com/NVIDIA/modulus/pull/544/files#diff-56b78f47e6e2657ca66fbed56b5c8a9ec69719f668eb7622c8855e119fa19143R30). Maybe instead of using `DistributedManager` inside of this function, I can pass rank as an argument to make it more clear?

Reply:
I think that would be great! Might make it modular too. :)

Reply:
Done.